openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -662,7 +681,7 @@ def _verify_user_unchanged(request, response, userid_in_session):
         #
 
         # The relevant views set a flag to indicate the exemption.
-        if getattr(response, 'safe_sessions_expected_user_change', None):
+        if CONTEXT.data.get('expected_user_change') is not None:
```

Comment:
I assume we still think it is safe to exempt the whole request, and not pay attention to specific user ids?

Reply:
Yes, although I'd love to instrument this at some point to see if we can make it stricter (which is why I have the `mark` function gathering the user ID even though it's not used yet.)